chat-sdk-ui/src/main/java/co/chatsdk/ui/threads/ThreadEditDetailsActivity.java

```diff
@@ -0,0 +1,162 @@
+            if (nameMetaValue != null)
+                nameInput.setText(nameMetaValue.getValue());
+        } else {
+            saveButton.setEnabled(false);
+        }
+
+        nameInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                saveButton.setEnabled(!nameInput.getText().toString().isEmpty());
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+
+            }
+        });
+
+        saveButton.setOnClickListener(v -> {
+            didClickOnSaveButton();
+        });
+
+    }
+
+    protected void didClickOnSaveButton() {
+        final String threadName = nameInput.getText().toString();
+        if (thread == null) {
+            showOrUpdateProgressDialog(getString(R.string.add_public_chat_dialog_progress_message));
+
+            disposableList.add(ChatSDK.publicThread().createPublicThreadWithName(threadName)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe((newThread, throwable) -> {
+                        if (throwable == null) {
+                            // TODO: permanently move thread name into meta data
+                            newThread.setMetaValue("name", threadName);
+                            disposableList.add(ChatSDK.thread().pushThreadMeta(newThread).subscribe(() -> {
+                                dismissProgressDialog();
+                                ToastHelper.show(ChatSDK.shared().context(), String.format(getString(R.string.thread__created), threadName));
+
+                                finish();
+                                ChatSDK.ui().startChatActivityForID(ChatSDK.shared().context(), newThread.getEntityID());
+                            }));
+                        } else {
+                            ChatSDK.logError(throwable);
+                            Toast.makeText(ChatSDK.shared().context(), throwable.getLocalizedMessage(), Toast.LENGTH_SHORT).show();
+                            dismissProgressDialog();
+                        }
+                    }));
+        } else {
+            // TODO: permanently move thread name into meta data
+            thread.setName(threadName);
+            thread.update();
+            thread.setMetaValue("name", threadName);
+            // TODO: Update the thread name
+            disposableList.add(ChatSDK.thread().pushThreadMeta(thread).subscribe(this::finish));
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        setResult(AppCompatActivity.RESULT_OK);
+
+        finish();
+        if (animateExit) {
+            overridePendingTransition(R.anim.dummy, R.anim.slide_top_bottom_out);
+        }
```

Comment:
Should this be after `finish()`

Reply:
Yes, I checked other activities which call `overridePendingTransition` and they first all call `finish()`

Reply:
Could you also comment this?